packages/api/src/modules/dataViz/amountsVsProgrammeRegion/amountsVSProgrammeRegion.service.test.ts

```diff
@@ -0,0 +1,179 @@
+import {
+    AMOUNTS_VS_PROGRAMME_REGION_ENTITIES,
+    NOT_AGGREGATED_ENTITIES,
+} from "./__fixtures__/amountsVSProgrammeRegion.fixture";
+import amountsVsProgrammeRegionPort from "../../../dataProviders/db/dataViz/amountVSProgrammeRegion/amountsVsProgrammeRegion.port";
+
+describe("amountsVSProgrammeRegionService", () => {
+    describe("toAmountsVsProgrammeRegionEntities", () => {
+        let mockAdapter: jest.SpyInstance;
+        let mockCursorFindChorusOnly: jest.SpyInstance;
+
+        let mockCursor;
+        let mockDocuments;
+        let nDocuments;
+
+        beforeEach(() => {
+            mockDocuments = [
+                PAYMENT_FLAT_ENTITY,
+                { ...PAYMENT_FLAT_ENTITY, amount: 7000 },
+                PAYMENT_FLAT_ENTITY_WITH_NULLS,
+            ];
+            nDocuments = mockDocuments.length;
+
+            mockCursor = {
+                next: jest.fn().mockImplementation(() => {
+                    return mockDocuments.shift();
+                }),
+                hasNext: jest.fn().mockImplementation(() => {
+                    if (mockDocuments.length) return true;
+                    return false;
+                }),
+            };
+
+            mockCursorFindChorusOnly = jest
+                .spyOn(paymentFlatService, "cursorFindChorusOnly")
+                .mockReturnValue(mockCursor);
+
+            mockAdapter = jest.spyOn(AmountsVsProgrammeRegionAdapter, "toNotAggregatedEntity");
+
+            mockAdapter.mockReturnValueOnce(NOT_AGGREGATED_ENTITIES[0]);
+            mockAdapter.mockReturnValueOnce(NOT_AGGREGATED_ENTITIES[2]);
+            mockAdapter.mockReturnValue(undefined);
+        });
+
+        afterEach(() => {
+            jest.resetAllMocks();
+        });
+
+        it("should call cursorFindChorusOnly with the budget exercice", async () => {
+            const exerciceBudgetaire = 2020;
+            await amountsVsProgrammeRegionService.toAmountsVsProgrammeRegionEntities(exerciceBudgetaire);
+
+            expect(mockCursorFindChorusOnly).toHaveBeenCalledWith(exerciceBudgetaire);
+        });
+
+        it("should call cursorFindChorusOnly without the budget exercice", async () => {
+            await amountsVsProgrammeRegionService.toAmountsVsProgrammeRegionEntities();
+
+            expect(mockCursorFindChorusOnly).toHaveBeenCalledWith(undefined);
+        });
+
+        it("should call hasNext nDocuments + 1 times", async () => {
+            await amountsVsProgrammeRegionService.toAmountsVsProgrammeRegionEntities();
+
+            expect(mockCursor.hasNext).toHaveBeenCalledTimes(nDocuments + 1);
+        });
+
+        it("should call cursor.next nDocuments times", async () => {
+            await amountsVsProgrammeRegionService.toAmountsVsProgrammeRegionEntities();
+
+            expect(mockCursor.next).toHaveBeenCalledTimes(nDocuments);
+        });
+
+        it("should call toNotAggregatedEntity amountsVsProgrammeRegion lenght", async () => {
```

Comment:
```suggestion
        it("should call toNotAggregatedEntity for each first time occurence of AmountsVsProgramRegion", async () => {
```

ou quelques chose comme ça, j'ai eu du mal à comprendre à la première lecture ce qu'était `amountsVsProgrammeRegion length`